plugins/HHAnalyzer.cc

```diff
@@ -801,6 +577,9 @@ void HHAnalyzer::analyze(const edm::Event& event, const edm::EventSetup&, const
             myllmetjj.DPhi_llmet_jj = fabs(ROOT::Math::VectorUtil::DeltaPhi(llmet[illmet].p4, jj[ijj].p4));
             myllmetjj.cosThetaStar_CS = fabs(getCosThetaStar_CS(llmet[illmet].p4, jj[ijj].p4));
             myllmetjj.MT_fullsystem = myllmetjj.p4.Mt();
+            myllmetjj.melaAngles = getMELAAngles(llmet[illmet].p4, jj[ijj].p4, leptons[ilep1].p4, leptons[ilep2].p4, jets[ijet1].p4, jets[ijet2].p4);
+            myllmetjj.visMelaAngles = getMELAAngles(ll[ill].p4, jj[ijj].p4, leptons[ilep1].p4, leptons[ilep2].p4, jets[ijet1].p4, jets[ijet2].p4); // only take the visible part of the H(ww) candidate
+
             // Some selection
             if (myllmetjj.minDR_l_j < m_minDR_l_j_Cut)
```

Comment:
I profit from the PR review to raise a concern about what has been done so far : we are only cleaning jets with the two leptons under consideration in the llmetjj candidate. I think it would be more consistent to clean them with *all* the leptons passing the ID under consideration (events with more than two leptons, which are not vetoed, can lead to jets being one of the extra lepton). The simple way would be to move lepton ID criteria in the electron/muon loop, and the minDR_l_j criteria in the jet loop (checking it for all leptons). This means that we do not have access to the other lepton ID collections "post-framework". Open for discussion.